bpf/lib/policy.h

```diff
@@ -24,9 +30,23 @@ __account_and_check(struct __ctx_buff *ctx __maybe_unused, struct policy_entry *
 		return DROP_POLICY_DENY;
 
 	*proxy_port = policy->proxy_port;
-	if (unlikely(policy->auth_type)) {
+
+	auth_type = policy->auth_type;
+	if (unlikely(!policy->has_explicit_auth_type &&
```

Comment:
Is there a clever way to achieve this with bitwise operations instead of comparisons?

Reply:
Not really, `has_explicit_auth_type` is the only bit there, `policy2` is a pointer and `auth_type` is a byte.